src/org/mozilla/javascript/Parser.java

```diff
@@ -2579,15 +2580,36 @@ private AstNode addExpr()
     private AstNode mulExpr()
         throws IOException
     {
-        AstNode pn = unaryExpr();
+        AstNode pn = expExpr();
         for (;;) {
             int tt = peekToken(), opPos = ts.tokenBeg;
             switch (tt) {
               case Token.MUL:
               case Token.DIV:
               case Token.MOD:
                 consumeToken();
-                pn = new InfixExpression(tt, pn, unaryExpr(), opPos);
+                pn = new InfixExpression(tt, pn, expExpr(), opPos);
+                continue;
+            }
+            break;
+        }
+        return pn;
+    }
+
+    private AstNode expExpr()
+        throws IOException
+    {
+        AstNode pn = unaryExpr();
+        for (;;) {
+            int tt = peekToken(), opPos = ts.tokenBeg;
+            switch (tt) {
+            case Token.EXP:
+                if (pn instanceof UnaryExpression) {
+                    reportError("msg.no.unary.expr.on.left.exp", AstNode.operatorToString(pn.getType()));
```

Comment:
```
js> -1 ** 2
js: "<stdin>", line 2: "-" is not allowed on the left-hand side of "**".
js: -1 ** 2
js: .....^
```
The error appears to be in the wrong position.
I tried `void reportError(String messageId, int position, int length)`, but it didn't change.

Reply:
I think that the shell does this for everything -- so fixing this shouldn't be in scope for this PR assuming that errors loaded from a regular file show up on the right line.